### DISCUSSION TITLE
RAM usage spikes

### DISCUSSION THREAD

**Discussion by tottero** [google/filament#4217] [Q&A]
In case of complex scenes when starting the rendering on Metal the RAM usage can increase quite dramatically but if the render process is paused the memory usage starts to decrease over time and it becomes constant. Is there a way to reduce the spike of the additional memory consumption during rendering by lowering cache sizes or similar in the engine?

**Comment by bejado** [comment]
Can you provide a scene (maybe a glTF file) that reproduces the issue? By pausing the render process, do you mean pausing calls to `beginFrame`, `render`, etc?

**Reply by tottero** [reply]
After investigating this a bit deeper the spike patterns in the RAM usage graph are most likely caused by the texture caching of the ResourceAllocator, and it is not related to a specific scene or backend.
In case of a more complex scene when we potentially run out of the texture cache size the memory usage starts to fluctuate and will have additional unexpected spikes which can easily reach the additional 500 MB+ territory. This might cause sudden out of memory issues on mobile devices. If the cache size is increased the spikes will disappear and the RAM usage will be close to a constant value which is close to the minimum of the range in the graph with the spikes.

**Reply by bejado** [reply]
@pixelflinger Maybe we should consider making the texture cache size configurable.

**Reply by pixelflinger** [reply]
Yes, this has come up before.  However, the memory would be kept high only if dynamic resolution was used and if it was very active. Otherwise it would quickly settle to the usage it needs. (Or if the viewport was changed every frame).
